Source/WebKit/Platform/cocoa/CocoaHelpers.mm

```diff
@@ -357,6 +363,17 @@ id parseJSON(API::Data& json, JSONOptionSet options, NSError **error)
     return [NSError errorWithDomain:error.domain ?: @"" code:error.code userInfo:nil].description;
 }
 
+NSURL *ensureDirectoryExists(NSURL *directory)
+{
+    ASSERT(directory.isFileURL);
+    if (!FileSystem::makeAllDirectories(directory.absoluteString)) {
```

Comment:
`absoluteString` will be incorrect, since it will contain `file:///` prefix.

`FileSystem::makeAllDirectories(directory.path)`

Reply:
This could also go in wtf/NSURLExtras.h

Reply:
This could also go in `wtf/NSURLExtras.mm/h`. Just need to drop the Extensions specific logging, which I would drop here if it stays anyway since CocoaHelpers is not extension specific.